packages/codecs-core/src/codec.ts

```diff
@@ -146,3 +142,61 @@ export function createCodec<From, To extends From = From>(
         write: codec.write,
     });
 }
+
+export function isFixedSizeCodec<T>(encoder: Encoder<T>): encoder is FixedSizeEncoder<T>;
+export function isFixedSizeCodec<T>(decoder: Decoder<T>): decoder is FixedSizeDecoder<T>;
+export function isFixedSizeCodec<T, U extends T = T>(codec: Codec<T, U>): codec is FixedSizeCodec<T, U>;
+export function isFixedSizeCodec<T, U extends T = T>(codec: Encoder<T> | Decoder<T> | Codec<T, U>): boolean {
+    return 'fixedSize' in codec && typeof codec.fixedSize === 'number';
+}
```

Comment:
Yeah I did this because essentially they were going to be duplicated functions. But I do like `isFixedSize` and `isVariableSize`! I'll append a new PR to the stack when I'm done with `codecs-data-structures` (that one's a big boi).

Reply:
Sounds good!

Reply:
Slight preference for a `const enum EncodingSize { FIXED, VARIABLE }`.

Reply:
Oh, I misread this thread. I thought the proposal was to add flags called `isFixedSize` and `isVariableSize`.

Carry on.